scipy/stats/tests/test_qmc.py

```diff
@@ -937,6 +938,36 @@ def test_fill_space(self):
         # circle packing problem is np complex
         assert l2_norm(sample) >= radius
 
+    def test_sample_inside_bounds(self):
+        radius = 0.2
+        l_bounds=[-1, -2, -1]
+        u_bounds=[3, 2, 1]
+        engine = self.qmce(d=3, radius=radius, l_bounds=l_bounds, u_bounds=u_bounds)
+        sample = engine.random(30)
+
+        for point in sample:
+            assert_array_less(point, u_bounds) 
+            assert_array_less(l_bounds, point) 
+
+    def test_inconsistent_bound_value(self):
+        radius = 0.2
+        l_bounds=[3, 2, 1]
+        u_bounds=[-1, -2, -1]
+        with pytest.raises(
+            ValueError, 
+            match="Bounds are not consistent 'l_bounds' < 'u_bounds'"):
+            self.qmce(d=3, radius=radius, l_bounds=l_bounds, u_bounds=u_bounds)
+
+    def test_inconsistent_bounds(self):
+        radius = 0.2
+        l_bounds=[-1, -2, -1]
+        u_bounds=[3, 2]
+        with pytest.raises(
+            ValueError, 
+            match="'l_bounds' and 'u_bounds' must be broadcastable and respect" 
+            " the sample dimension"):
+            self.qmce(d=3, radius=radius, l_bounds=l_bounds, u_bounds=u_bounds)
```

Comment:
Also maybe add a test which has 2 valid dims for the bounds but in qmce uses `d=3`. This should also fail.

Reply:
Thanks for the update, if the test is that similar, can you then parametrize it.

Reply:
you mean the bounds values, right?

Reply:
Yes, your update looks good thank you